shared/githubconnector/download.go

```diff
@@ -0,0 +1,62 @@
+package githubconnector
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path"
+)
+
+func (client *GithubConnector) DownloadPackage(org, project, version, os, arch, path string) (string, error) {
+
+	release, err := client.GetRelease(org, project, version)
+
+	if err != nil {
+		fmt.Println("Error getting release: ", err)
+		return "", err
+	}
+
+	for _, asset := range release.Assets {
+		if asset.Component == "happy" && asset.OS == os && asset.Architecture == arch {
+			return download(asset.URL, asset.Name, path)
+		}
+	}
+
+	return "", errors.New("no suitable package found")
+}
+
+func download(url, fileName, dir string) (string, error) {
+
+	filePath := path.Join(dir, fileName)
+	// Download the url to the specified path/fileName
+	file, err := os.Create(filePath)
+	if err != nil {
+		return "", err
+	}
+
+	client := http.Client{
+		CheckRedirect: func(r *http.Request, via []*http.Request) error {
+			r.URL.Opaque = r.URL.Path
+			return nil
+		},
+	}
+
+	resp, err := client.Get(url)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
let's bubble up this error, instead of using fatal. I think fatal causes the program to panic right here